src/Http/Controllers/Detect/DetectPatternController.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace TanHongIT\LaravelGenerator\Http\Controllers\Detect;
+
+use Illuminate\Routing\Controller;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use ReflectionClass;
+use ReflectionMethod;
+
+class DetectPatternController extends Controller
```

Comment:
This is a class to check if the current repository pattern is used in the application. It only checks relatively, so it is not always accurate depending on the source structure.